src/copyAs.ts

```diff
@@ -0,0 +1,118 @@
+import * as base64 from "js-base64";
+import * as vscode from "vscode";
+import { CopyFormat, ExtensionHostMessageHandler, MessageType } from "../shared/protocol";
+
+interface QuickPickCopyFormat extends vscode.QuickPickItem {
+	label: CopyFormat;
+}
+
+export const copyAs = async (messaging: ExtensionHostMessageHandler): Promise<void> => {
+	const formats: QuickPickCopyFormat[] = [
+		{ label: CopyFormat.Hex },
+		{ label: CopyFormat.Literal },
+		{ label: CopyFormat.Utf8 },
+		{ label: CopyFormat.C },
+		{ label: CopyFormat.Go },
+		{ label: CopyFormat.Java },
+		{ label: CopyFormat.JSON },
+		{ label: CopyFormat.Base64 },
+	];
+
+	vscode.window.showQuickPick(formats).then(format => {
+		if (format) {
+			messaging.sendEvent({ type: MessageType.TriggerCopyAs, format: format["label"] });
+		}
+	});
+};
+
+export function copyAsText(buffer: Uint8Array) {
+	vscode.env.clipboard.writeText(new TextDecoder().decode(buffer));
+}
+
+export function copyAsHex(buffer: Uint8Array) {
+	vscode.env.clipboard.writeText(Buffer.from(buffer).toString("hex"));
+}
+
+export function copyAsLiteral(buffer: Uint8Array) {
+	let encoded: string = "";
+	const digits = Buffer.from(buffer)
+		.toString("hex")
+		.match(/.{1,2}/g);
+	if (digits) {
+		encoded = "\\x" + digits.join("\\x");
+	}
+
+	vscode.env.clipboard.writeText(encoded);
+}
+
+export function copyAsC(buffer: Uint8Array, filename: string) {
+	const len = buffer.length;
+	let content: string = `unsigned char ${filename}[${len}] =\n{`;
```

Comment:
In these files you will want to make sure the filename is somewhat valid. Some very basic regex that replaces all non-alphanumeric characters is probably good enough, `filename.replace(/[^a-z0-9]/ig, '')`